src/settings/window_geometry.rs

```diff
@@ -1,16 +1,73 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
-pub const DEFAULT_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
+pub fn try_to_load_last_window_size() -> Result<WindowGeometry, String> {
```

Comment:
I think it's better to rename this to `maybe_load_saved_window_size`.

Reply:
It has different semantics from `maybe_save_window_size` though. There, we either save or not save, whereas in `try_to_load_last_window_size` we always load, except that the loading may fail

Reply:
But it's better to make it behave just as `maybe_save_window_size`, check the flag, maybe load.

Reply:
Ohh so while that would be ideal, the flag won't be set before this function gets called, as you have mentioned in our discussion below.